### PR TITLE
feat: restructure repository and add integrations

### PR DESCRIPTION
**Pull request by jrriehl**
- Move Python library into `python` folder
- Update README files
- Add `integrations` folder with initial `fetch-holiday` integration

### CODE REVIEW

File: .github/workflows/ci-tests.yml [hunk 1]
```diff
@@ -15,6 +15,7 @@ jobs:
     defaults:
```
Comment (lrahmani) [inline .github/workflows/ci-tests.yml:15]: You would want to add `path` to `on: push:` to run the workflow only when files in `python` are updated

Comment (jrriehl) [reply .github/workflows/ci-tests.yml:15]: Do you think we need on `push` if we also run this on `pull_request`?

**Comment by GataKamsky** [comment]
Looks good from my perspective. @ejfitzgerald you good with this?

**Review by GataKamsky — APPROVED** [review]
Good from my perspective.

File: .github/workflows/ci-tests.yml [hunk 2]
```diff
@@ -1,20 +1,20 @@
 name: CI
 
 on:
-  push:
-    branches:
-      - main
-
```
Comment (ejfitzgerald) [inline .github/workflows/ci-tests.yml:7]: Would still keep this. Worth running full checks on each commit to `main`

**Review by ejfitzgerald — COMMENTED** [review]
Looks good one minor CI comment

**Review by ejfitzgerald — APPROVED** [review]

**Pull request merged**